P1_BMI_Stats.R

```diff
@@ -256,65 +252,399 @@ bmiData <- bmiData %>%
+#      'simd2009v2_sc_quintile', 'simd2012_sc_quintile', 'simd2012_sc_quintile', 'simd2012_sc_quintile',
+#      'simd2012_sc_quintile', 'simd2016_sc_quintile', 'simd2016_sc_quintile', 'simd2016_sc_quintile',
+#      'simd2016_sc_quintile')
+# bmi_data < bmi_data %>%
+#   for (schlyr_exam in a) {
+#     allsimdfinyear <- b}
+mutate(simd = allsimdfinyear),
+if is.na(allsimdfinyear) simd='U', #x of n records don't have a SIMD quintile.
+
+# Councils - create the old CA with codes 01-32 ; the new codes are made unique by the last 2 chars so use them   .
+bmi_data < bmi_data %>%
+  for (CA in c("33", "34", "41", "35", "26", "05", "39", "06", "42", "08", "45", "10", "11", "36", "14", "47", "46", "17", "18", "19", "20", "21", "44", "23", "48", "38", "27", "28", "29", "30", "40", "13")) {
+       CA <- c("01", "02", "03", "04", "05", "06", "07", "08", "09", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19", "20", "21", "22", "23", "24", "25", "26", "27", "28", "29", "30", "31", "32")}, 
+mutate(council_area_name <- sapply(council_area_name,rescale)), 
+# CA_name = case_when(CA == 1 ~ "Aberdeen City",
+#                     CA == 2 ~ "Aberdeenshire",
+#                     CA == 3 ~ "Angus", 
+#                     CA == 4 ~ "Argyll & Bute",
+#                     CA == 5 ~ "Scottish Borders", 
+#                     CA == 6 ~ "Clackmannanshire",
+#                     CA == 7 ~ "West Dunbartonshire", 
+#                     CA == 8 ~ "Dumfries & Galloway",
+#                     CA == 9 ~ "Dundee City", 
+#                     CA == 10 ~ "East Ayrshire",
+#                     CA == 11 ~ "East Dunbartonshire", 
+#                     CA == 12 ~ "East Lothian",
+#                     CA == 13 ~ "East Renfrewshire", 
+#                     CA == 14 ~ "City of Edinburgh",
+#                     CA == 15 ~ "Falkirk",
+#                     CA == 16 ~ "Fife",
+#                     CA == 17 ~ "Glasgow City", 
+#                     CA == 18 ~ "Highland",
+#                     CA == 19 ~ "Inverclyde",
+#                     CA == 20 ~ "Midlothian",
+#                     CA == 21 ~ "Moray",
+#                     CA == 22 ~ "North Ayrshire",
+#                     CA == 23 ~ "North Lanarkshire", 
+#                     CA == 24 ~ "Orkney Islands",
+#                     CA == 25 ~ "Perth & Kinross",
+#                     CA == 26 ~ "Renfrewshire",
+#                     CA == 27 ~ "Shetland Islands",
+#                     CA == 28 ~ "South Ayrshire",
+#                     CA == 29 ~ "South Lanarkshire",
+#                     CA == 30 ~ "Stirling",
+#                     CA == 31 ~ "West Lothian",
+#                     CA == 32 ~ "Comhairle nan Eilean Siar",
+#                     TRUE ~ "Other"),
+mutate carea=paste("CA",CA),
+
+
+arrange(bmi_data, chi)
+
+
+bmi_basefile <- bmi_data %>%
+  subset(select = c(chi id HB2018 CA2018 CA carea sex1 height weight daterec schlyr_exam schlgivn rev_num agemth nyob mob dob PC7
```

Comment:
I have significantly reduced the number of variables that we keep for the basefile.